### PR TITLE
Activate windows build

### PR DESCRIPTION
**Pull request by dbast**
Checklist
* [x] Used a fork of the feedstock to propose changes
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by dbast** [comment]
Seems to require more work... A win build could be possible now as there was some upstream work, see e.g. https://github.com/apache/arrow/commit/1d588ab35241792f2e702e90ba331e9dc659354a

File: recipe/meta.yaml [hunk 1]
```diff
@@ -12,23 +12,27 @@ source:
...
   rpaths:
     - lib/R/lib/
     - lib/
 
 requirements:
   build:
-    - {{ compiler('c') }}        # [not win]
-    - {{ compiler('cxx') }}      # [not win]
+    - {{ compiler('c') }}              # [not win]
+    - {{ compiler('m2w64_c') }}        # [win]
```
Comment (xhochy) [inline recipe/meta.yaml]: Is there a reason we don't want to use the default compiler on Windows?

Comment (dbast) [reply recipe/meta.yaml]: This is a reference to m2w64-toolchain, which includes the default compiler for R recipes.

Comment (xhochy) [reply recipe/meta.yaml]: As far as a I understood this is only used as a lot of R recipes don't compile with the default compilers but isn't a requirement? My guess is that the build errors go away when using the default compilers.

Comment (xhochy) [reply recipe/meta.yaml]: Now I understand better: https://ci.appveyor.com/project/conda-forge/r-arrow-feedstock/builds/25955436/job/klrd07nivr8ri90j :)

**Comment by dhirschfeld** [comment]
It appears that Windows builds are still disabled?
https://github.com/conda-forge/r-arrow-feedstock/blob/3014741c4ce088737e77e0e0ecbdc5d377a85c5c/recipe/meta.yaml#L16

Are there still some blockers preventing building a Windows package?

**Comment by dbast** [comment]
@dhirschfeld Right, the windows build is still deactivated. The idea of this PR was to remove the `skip: True  # [win]` and see if the package just would build. But is does not and further investigations seem to be required. Feel free to open a PR to work on this, maybe the build errors of this PR and the mentioned links help to move forward.

**Comment by dhirschfeld** [comment]
For ref there are a bunch of warnings:
```
bash.exe: warning: could not find /tmp, please create!
```
...and then compilation fails with:
```
g++ -shared -s -o arrow.dll tmp.def array.o array__to_vector.o array_from_vector.o arraydata.o arrowExports.o buffer.o chunkedarray.o column.o compression.o compute.o csv.o datatype.o feather.o field.o io.o json.o memorypool.o message.o parquet.o recordbatch.o recordbatchreader.o recordbatchwriter.o schema.o symbols.o table.o threadpool.o -L../windows/arrow-0.14.0/lib-5.3.0/x64 -L../windows/arrow-0.14.0/lib/x64 -lparquet -larrow -lthrift -lsnappy -lboost_regex-mt-s -lboost_filesystem-mt-s -lboost_system-mt-s -ldouble-conversion -lz -lws2_32 -LC:/bld/r-arrow_1562913776224/_h_env/lib/R/../../Library/mingw-w64/lib/x64 -LC:/bld/r-arrow_1562913776224/_h_env/lib/R/../../Library/mingw-w64/lib -LC:/bld/r-arrow_1562913776224/_h_env/lib/R/bin/x64 -lR
C:/bld/r-arrow_1562913776224/_h_env/Library/mingw-w64/bin/../lib/gcc/x86_64-w64-mingw32/5.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: cannot find -lparquet
C:/bld/r-arrow_1562913776224/_h_env/Library/mingw-w64/bin/../lib/gcc/x86_64-w64-mingw32/5.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: cannot find -larrow
C:/bld/r-arrow_1562913776224/_h_env/Library/mingw-w64/bin/../lib/gcc/x86_64-w64-mingw32/5.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: cannot find -lthrift
C:/bld/r-arrow_1562913776224/_h_env/Library/mingw-w64/bin/../lib/gcc/x86_64-w64-mingw32/5.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: cannot find -lsnappy
C:/bld/r-arrow_1562913776224/_h_env/Library/mingw-w64/bin/../lib/gcc/x86_64-w64-mingw32/5.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: cannot find -lboost_regex-mt-s
C:/bld/r-arrow_1562913776224/_h_env/Library/mingw-w64/bin/../lib/gcc/x86_64-w64-mingw32/5.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: cannot find -lboost_filesystem-mt-s
C:/bld/r-arrow_1562913776224/_h_env/Library/mingw-w64/bin/../lib/gcc/x86_64-w64-mingw32/5.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: cannot find -lboost_system-mt-s
C:/bld/r-arrow_1562913776224/_h_env/Library/mingw-w64/bin/../lib/gcc/x86_64-w64-mingw32/5.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: cannot find -ldouble-conversion
collect2.exe: error: ld returned 1 exit status
/usr/bin/rm: missing operand
Try '/usr/bin/rm --help' for more information.
make: *** [arrow.dll] Error 1
C:/bld/r-arrow_1562913776224/_h_env/lib/R/share/make/winshlib.mk:13: recipe for target 'arrow.dll' failed
ERROR: compilation failed for package 'arrow'
* removing 'C:/bld/r-arrow_1562913776224/_h_env/Lib/R/library/arrow'
In R CMD INSTALL
```

**Pull request closed without merging**